Add SSO integration guide for Apache Airflow

1abdelhalim opened the pull request:
## Add comprehensive SSO integration guide for Apache Airflow

This PR adds comprehensive documentation for integrating Single Sign-On (SSO) in Apache Airflow, addressing issue #55276.

### What's Included:
- **Step-by-step SSO setup guide** using FAB Auth Manager
- **Multiple provider examples**: Generic OAuth2, Okta, Azure Entra ID, Google
- **Two configuration approaches**: Environment variables and config file
- **Comprehensive troubleshooting section** with common issues and solutions
- **Cross-references to provider-specific documentation**

### Changes Made Based on Reviewer Feedback:
- **Correct auth manager info**: SimpleAuthManager is default in Airflow 3
- **Added provider examples**: Okta, Azure Entra ID, Google as requested
- **Airflow 3 compatibility**: Removed version-specific references
- **Added cross-references**: Links to provider docs per @eladkal feedback
- **Moved to FAB provider**: Correct location per maintainer feedback

### Additional Fixes:
Also includes fixes for GitHub Actions CI issues encountered during development:
- Fixed git revision syntax errors (`...` → `..`)
- Added dependency checks for `gh` and `jq` tools

### Files Added:
- `providers/fab/docs/auth-manager/sso.rst` - SSO integration guide
- Updated FAB auth manager index and spelling wordlist

### Addresses:
- Issue #55276
- CI workflow improvements

### Checklist:
- [x] Documentation added
- [x] Airflow 3 compatible
- [x] CI issues resolved

vincbeck commented on airflow-core/docs/security/sso_example.rst:
```diff
@@ -0,0 +1,81 @@
+
+Prerequisites
+-------------
+- Airflow 2.x installed and running
```
Is it for Airflow 2 only? If so, it should be in the docs, whether we put here is for Airflow 3. As a matter of fact we do not mention the Airflow version in docs, we just mention `Airflow`. In Airflow doc website, the user can pick the version with the top left dropdown.

vincbeck commented on airflow-core/docs/security/sso_example.rst:
```diff
@@ -0,0 +1,81 @@
+
+1. **Install Required Packages**
+   
+   Airflow uses Flask AppBuilder (FAB) for authentication. To enable OAuth, install the required provider:
```
This is not true. Airflow delegates authentication to auth manager. These auth managers, which can be viewed as plugins, are responsible for authentication and authorization. I think you are referring to Fab auth manager. In that case, I'd advise you to create this doc in the Fab provider.

jroachgolf84 replied:
I also don't believe that the FAB Auth Manager is the default auth manager anymore... It needs to be specified, otherwise I believe it uses the Simple Auth Manager.

vincbeck commented on airflow-core/docs/security/sso_example.rst:
```diff
@@ -0,0 +1,81 @@
+
+   .. code-block:: ini
+
+      [webserver]
```
This command no longer exist

vincbeck reviewed (changes requested):
Blocking this PR, it uses a lot of Airflow 2 code

jroachgolf84 commented on airflow-core/docs/security/sso_example.rst:
```diff
@@ -0,0 +1,81 @@
+
```
I think that this is quite useful, but should include specific examples as well (Okta, Entra ID, etc.).

vincbeck reviewed (commented):
This documentation needs to be in FAB provider and not Airflow documentation. This documentation relates to Fab auth manager. Fab provider has already some documentation about authentication, it will perfectly fit next to them. Example of existing documentation:
- https://airflow.apache.org/docs/apache-airflow-providers-fab/stable/auth-manager/api-authentication.html
- https://airflow.apache.org/docs/apache-airflow-providers-fab/stable/auth-manager/webserver-authentication.html

In the codebase, Fab provider documentation is available in `providers/fab/docs/`

eladkal commented on providers/fab/docs/auth-manager/sso.rst:
```diff
@@ -0,0 +1,228 @@
+Prerequisites
+-------------
+- Apache Airflow installed and running with FAB Auth Manager
+- Access to an OAuth2 SSO provider (e.g., Google, Okta, Auth0, Azure Entra ID)
```
Do we need to cross reference to the docs in these specific providers that are about sso?

1abdelhalim replied:
Good one. 
Added cross-references to the provider-specific docs in Prerequisites and each example section. Should help users find the setup instructions they need 👍

1abdelhalim committed:
Add SSO documentation to FAB auth manager

- Add comprehensive SSO documentation with OAuth2 provider examples
- Include configuration for Google, Okta, Azure Entra ID, and generic providers
- Add troubleshooting section and references
- Update FAB auth manager index to include SSO documentation

This addresses the need for clear SSO configuration guidance in the FAB provider.

1abdelhalim committed:
CI: install gh and jq early in ci-amd; docs: strip trailing spaces in SSO doc

eladkal commented on .github/workflows/ci-amd.yml:
```diff
@@ -139,6 +139,10 @@ jobs:
+      - name: "Install gh and jq"
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y gh jq
```
There should be no changes to workflows with this PR

1abdelhalim committed:
Revert workflow changes per review: no workflow edits in docs PR

1abdelhalim committed:
docs(spellcheck): allow ‘Entra’ in Sphinx spell list

1abdelhalim committed:
docs(fab): fix RST title overline; spellcheck: keep ‘Entra’ sorted in wordlist

1abdelhalim committed:
docs(fab): inline spelling allow for ‘Entra’ to satisfy spellcheck

vincbeck commented on docs/spelling_wordlist.txt:
```diff
@@ -606,6 +606,7 @@ encodable
 encryptor
 enqueue
 enqueued
+Entra
```
Twice the same word?

vincbeck commented on providers/fab/docs/auth-manager/sso.rst:
```diff
@@ -0,0 +1,251 @@
+
+.. spelling::
+   word-list::
+     Entra
```
No need

1abdelhalim replied:
Ok, thank you so much 
I will remove this one 
is there any other suggestions to make this pr ready to be merged ?

vincbeck replied:
Now it is good

vincbeck reviewed (changes requested):
Updating my approval to request changes. Using AI to create PR is totally okay/recommended. However, reviewing the code generated and making sure the quality is good enough to create/update a PR is also very recommended.

1abdelhalim committed:
docs(spellcheck): keep only one ‘Entra’ entry; remove inline spelling directive

1abdelhalim committed:
docs(spellcheck): re-add 'Entra' to spelling wordlist - needed for Azure Entra ID references

1abdelhalim committed:
Merge branch 'main' into docs/sso_example

vincbeck committed:
Merge branch 'main' into docs/sso_example

1abdelhalim committed:
Merge branch 'main' into docs/sso_example
